File: app/src/main/java/com/MohamedTaha/Imagine/New/Adapter/AdapterElarbaoonElnawawy.java
package com.MohamedTaha.Imagine.New.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.MohamedTaha.Imagine.New.R;
import com.MohamedTaha.Imagine.New.databinding.CustomElarbaoonElnawawyBinding;
import com.MohamedTaha.Imagine.New.mvp.model.ElarbaoonElnawawyModel;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class AdapterElarbaoonElnawawy extends RecyclerView.Adapter<AdapterElarbaoonElnawawy.ElarbaoonViewHolder> {

    private CustomElarbaoonElnawawyBinding customElarbaoonElnawawyBinding;
    private List<ElarbaoonElnawawyModel> elnawawyModelsList;
    private Context context;

    private ClickListener clickListener;

    public interface ClickListener {
        void onClick(int position);
    }

    public AdapterElarbaoonElnawawy(List<ElarbaoonElnawawyModel> elnawawyModelsList, ClickListener clickListener) {
        this.elnawawyModelsList = elnawawyModelsList;
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public ElarbaoonViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        //customElarbaoonElnawawyBinding = CustomElarbaoonElnawawyBinding.inflate();
        View row = LayoutInflater.from(parent.getContext()).inflate(R.layout.custom_elarbaoon_elnawawy, null);
        ElarbaoonViewHolder elarbaoonViewHolder = new ElarbaoonViewHolder(row);
        row.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int positionRow = elarbaoonViewHolder.getAdapterPosition();
                if (clickListener != null) clickListener.onClick(positionRow);
            }
        });

        return elarbaoonViewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ElarbaoonViewHolder holder, int position) {
        ElarbaoonElnawawyModel elnawawyModel = elnawawyModelsList.get(position);
        holder.TVNameElhadeth.setText(elnawawyModel.getName_elhadeth());
        holder.TVNumberElhadeth.setText(elnawawyModel.getNumber_elhadeth());

    }

    @Override
    public int getItemCount() {
        return elnawawyModelsList.size();
    }

    static class ElarbaoonViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.TV_Name_Elhadeth)
        TextView TVNameElhadeth;
        @BindView(R.id.TV_Number_Elhadeth)
        TextView TVNumberElhadeth;
        public ElarbaoonViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);

        }
    }
}

File: app/src/main/java/com/mohamedtaha/imagine/ui/navigationview/adapter/AdapterElarbaoonElnawawy.kt
package com.mohamedtaha.imagine.ui.navigationview.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.mohamedtaha.imagine.databinding.CustomElarbaoonElnawawyBinding
import com.mohamedtaha.imagine.mvp.model.ElarbaoonElnawawyModel
import com.mohamedtaha.imagine.ui.navigationview.adapter.AdapterElarbaoonElnawawy.ElarbaoonViewHolder
import com.mohamedtaha.imagine.util.ClickListener

class AdapterElarbaoonElnawawy(
    private val clickListener: ClickListener<ElarbaoonElnawawyModel>
) : RecyclerView.Adapter<ElarbaoonViewHolder>() {
    private lateinit var elnawawyModelsList: List<ElarbaoonElnawawyModel>
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ElarbaoonViewHolder {
        val viewHolder = ElarbaoonViewHolder(
            CustomElarbaoonElnawawyBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
        viewHolder.itemView.setOnClickListener {
            val item = elnawawyModelsList[viewHolder.adapterPosition]
            clickListener.onClick(it, item)
        }
        return viewHolder
    }
    fun setData(elnawawyModelsList: List<ElarbaoonElnawawyModel>){
        this.elnawawyModelsList = elnawawyModelsList
    }

    override fun onBindViewHolder(holder: ElarbaoonViewHolder, position: Int) {
        val elnawawyModel = elnawawyModelsList[position]
        holder.bind(elnawawyModel)
    }

    override fun getItemCount(): Int {
        return elnawawyModelsList.size
    }

    class ElarbaoonViewHolder(val binding: CustomElarbaoonElnawawyBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun bind(elarbaoonElnawawyModel: ElarbaoonElnawawyModel) {
            binding.elarbaoonElnawawy = elarbaoonElnawawyModel
            binding.executePendingBindings()
        }
    }
}